### ISSUE TITLE
Bug in execute function where only the first stderr is ever captured

### ISSUE THREAD

**Issue by dkosteck** [redhat-partner-solutions/intel-sriov-test#78]
For instance, if you try to run the command "testing" as the first command run by the dut you will receive the following in outs, errs: 
`INTERNALERROR> AssertionError: []
INTERNALERROR> [['-bash: testing: command not found\n']]`

If it is the second command run it will be the following in outs, errs:
`INTERNALERROR> AssertionError: []
INTERNALERROR> [[]]`

**Comment by dkosteck** [comment]
This is a very subtle timing issue, documentation to follow

**Comment by dkosteck** [comment]
The current code which is causing the issues begins here: https://github.com/redhat-partner-solutions/intel-sriov-test/blob/edace23ed3e395e069927ddc169f9f2502b6aa66/sriov/common/exec.py#L205

Essentially the order in which commands are executed are as follows:
```
write cmd to stdin
write echo to stdin
flush stdin
```
The expected stdout is as follows:
```
cmd
cmd stdout
echo
echo stdout
```
However, the observed stdout is as follows:
```
cmd
echo
cmd stdout
echo stdout
```
This can be explained by the fact we are just reading in the buffered stdout from the terminal, on which two commands were run asynchronously. 
Thus, we expect the sherr list in our execute command to look like the following (at the moment it finishes parsing the stdout):
```
['terminal> cmd', 'cmd error', 'echo cmd']
```
 But the following is a more representative example (because of the ordering of the results, the stdout buffer is reset after the original terminal command is in it, causing it to be lost):
```
['cmd error', 'echo cmd'] 
```
The clear issue in this case is that because there are only two items in the list, the latter of which will be popped by the logic to remove the echo command, and the first element of which may be popped by the logic to pop the original command (as the error may contain the original name of the command in it, a real world example: `[root@flexran-ru tests]# testing 
-bash: testing: command not found`). This will result in an empty stderr, as observed.

There seem to be two potential remedies to this:
1. Introducing some delay between running the command and the echo, hoping to force correct ordering of stdout.
2. Only popping the initial element of the sherr list if the length of the list is greater than 1. 

The latter of these solutions seems to cover the most cases, and should be resilient to either of the orderings of stdout (except a situation where the echo result would arrive first, in which case the function would not be able to correctly get the exit code and would fail).

**Comment by jianzzha** [comment]
@dkosteck This is a good finding. In your test, did you ever see it can hit this condition?
https://github.com/redhat-partner-solutions/intel-sriov-test/blob/edace23ed3e395e069927ddc169f9f2502b6aa66/sriov/common/exec.py#L218

**Comment by dkosteck** [comment]
> @dkosteck This is a good finding. In your test, did you ever see it can hit this condition?
> 
> https://github.com/redhat-partner-solutions/intel-sriov-test/blob/edace23ed3e395e069927ddc169f9f2502b6aa66/sriov/common/exec.py#L218

Yes, I did see this when running it from the flexran-ru node, again we've seen some other timing things going from this machine to the DUTs. At one point when debugging tests I noticed that the errs list was empty when it should have contained something from stderr

**Comment by dkosteck** [comment]
Fixed as of #81
